### ISSUE TITLE
Apollo GraphQL doesn't use new variables after refetching!

### ISSUE THREAD

**Issue by makamekm** [apollographql/apollo-client#2490]
Again... GraphQL doesn't use new variables after refetching.
```
@graphql<Api.LoadLTorEqQuery, Api.LoadLTorEqQueryVariables, {
  dataProps: DataProps
} & any>(Api.QL.Main.LoadLTorEq, {
  props: ({ ownProps, data: { loading, loadLTorEq, refetch } }) => ({
    data: loadLTorEq,
    loadingData: loading,
    refetchData: refetch
  }),
  options: (props) => ({
    variables: {
      timestamp: props.dataProps.date.getTime(),
      limit: props.dataProps.limit,
      symbols: props.allCoins.slice(props.pagination.offset, props.pagination.limit + props.pagination.offset),
      timeframe: props.dataProps.timefame
    }
  }),
})
```
`symbols` are always the same.

This is a duplication of an issue: [Issue](https://github.com/apollographql/react-apollo/issues/1321)

**Comment by msmfsd** [comment]
Bump, here is what happens on v2 only:
```
1. turn server offline
2. component loads with error { loading: false, error: 'Error: Network error: Failed to fetch', myUser: undefined  }
3. turn server online and refetch
4. component loads data successfully but error is not cleared { loading: false, error: 'Error: Network error: Failed to fetch', myUser: { uuid: GJGHJ86876HUIIUH }  }
```

**Comment by jbaxleyiii** [comment]
Thanks for the reproduction! I'll take a look!

**Comment by ex3ndr** [comment]
I have some similar problem. If i will NOT provide some variable and then (in other query) remove it then graphql doesn't perform query.

**Comment by hwillson** [comment]
This sounds like the same problem as https://github.com/apollographql/apollo-client/issues/2499, which was resolved in https://github.com/apollographql/apollo-client/pull/3500, and released in `apollo-client` 2.3.2. Closing - thanks all!
